wallet-new/src/Cardano/Wallet/Kernel/DB/Util/IxSet.hs

```diff
@@ -83,6 +99,15 @@ newtype IxSet a = WrapIxSet {
       unwrapIxSet :: IxSet.IxSet (PrimKey a ': IndicesOf a) (OrdByPrimKey a)
     }
 
+instance Show a => Show (IxSet a) where
+    show = show . map unwrapOrdByPrimKey . IxSet.toList . unwrapIxSet
+
+instance (Indexable a, HasPrimKey a) => Eq (IxSet a) where
+    (WrapIxSet ix1) == (WrapIxSet ix2) = ix1 == ix2
+
+instance (Indexable a, HasPrimKey a) => Ord (IxSet a) where
+    compare (WrapIxSet ix1) (WrapIxSet ix2) = compare ix1 ix2
```

Comment:
Not a fan of these instances. Do we really want to compare _entire_ `IxSet`s??

Reply:
I suspect these were needed to abide to some other upstream type/function requiring it. Surely I can try to rework this instance, but this begs the perhaps silly question: if I call `compare` on the underlying `IxSet`, shouldn't I expect the implementation to short circuit and to be logarithmic anyway? If I need to keep this around, can you think of any better implementation?

Reply:
You've lost me. How could full comparison possibly be logarithmic?

Reply:
I don't want you to rework the instance, I don't want the instance at all :)